src/maker/api.rs

```diff
@@ -126,9 +126,30 @@ impl ThreadPool {
             .threads
             .lock()
             .map_err(|_| MakerError::General("Failed to lock threads"))?;
+
+        let thread_count = threads.len();
+        log::info!("Joining {} threads", thread_count);
+
+        let mut joined_count = 0;
         while let Some(thread) = threads.pop() {
-            thread.join().unwrap();
+            let thread_name = thread.thread().name().unwrap().to_string();
+
+            match thread.join() {
+                Ok(_) => {
+                    log::info!("Thread {} terminated successfully", thread_name);
```

Comment:
Would be helpful if we could add the port number to this log. Allows us to see which maker this thread belongs too. There could be many ways to do it. Easiest would be to add a `port: u32` field in the `Threadpool`, and use that in all logs.

Reply:
nit: 
``` suggestion 
                    log::info!("Thread {} joined", thread_name);
```